frontend/src/modules/auth/store/auth.actions.ts

```diff
@@ -0,0 +1,94 @@
+import { Auth0Service } from '@/modules/auth/services/auth0.service';
+import { AuthApiService } from '@/modules/auth/services/auth.api.service';
+import { AuthService } from '@/modules/auth/services/auth.service';
+import { User } from '@/modules/auth/types/User.type';
+import Errors from '@/shared/error/errors';
+import { disconnectSocket, connectSocket } from '@/modules/auth/auth.socket';
+import identify from '@/shared/monitoring/identify';
+import { watch } from 'vue';
+
+export default {
+  init() {
+    Auth0Service.isAuthenticated()
+      .then((isAuthenticated: boolean) => {
+        if (!isAuthenticated) {
+          Auth0Service.getTokenSilently()
+            .then(() => Auth0Service.authData())
+            .then((token) => this.authCallback(token))
+            .catch((error) => {
+              if (['login_required', 'consent_required', 'missing_refresh_token'].includes(error.error)) {
+                const appState: any = {};
+                if (window.location.hash) {
+                  // Store the current virtual path or other data into appState to
+                  // retrieve when handling a redirect callback.
+                  appState.returnTo = window.location.hash;
+                }
+                return this.signin();
+              }
+              return Promise.reject();
+            });
+        }
+      });
+  },
+  ensureLoaded(): Promise<void> {
+    if (!this.user || !this.tenant) {
+      return new Promise((resolve) => {
+        const stopWatcher = watch(
+          () => [this.user, this.tenant],
+          ([newUser, newTenant]) => {
+            if (newUser && newTenant) {
+              resolve();
+              stopWatcher();
+            }
+          },
+          {
+            immediate: true,
+          },
+        );
+      });
+    }
+    // Both are already loaded
+    return Promise.resolve();
+  },
+  getUser(token?: string) {
```

Comment:
Should you be connecting the socket every time you call getUser?
I think you are calling getUser in some places of the app so that it will make a new connection every time you call this

Reply:
True, thanks for notice, fixing